### PR TITLE
Ga provisioning/data endpoint followup

### PR DESCRIPTION
**Pull request by adamsilverstein**
## Summary

Addresses issue https://github.com/google/site-kit-wp/issues/1212

## Relevant technical choices

Addresses final feedback on issue.

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: includes/Modules/Analytics/proxy-service-analytics-provisioning.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Class Google\Site_Kit\Modules\Analytics\Google_Service_AnalyticsProvisioning
```
Comment (felixarntz) [inline includes/Modules/Analytics/proxy-service-analytics-provisioning.php:3]: This file name doesn't match PSR-4 (it must match the class name, like elsewhere), so this needs to be changed. If we didn't have the classmap, this name would throw a fatal error.

File: includes/Modules/Analytics/proxy-service-analytics-provisioning.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * Class for Analytics Provisioning service.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Google_Service_AnalyticsProvisioning extends Google_Service_Analytics {
+	/**
+	 * Constructs the internal representation of the Analytics service.
+	 *
```
Comment (felixarntz) [inline includes/Modules/Analytics/proxy-service-analytics-provisioning.php:26]: Missing since tag here.

File: includes/Modules/Analytics.php [hunk 3]
```diff
@@ -662,6 +663,54 @@ protected function create_data_request( Data_Request $data ) {
...
+					/* translators: %s: Missing parameter name */
+					return new WP_Error( 'missing_required_param', sprintf( __( 'Request parameter is empty: %s.', 'google-site-kit' ), 'profileName' ), array( 'status' => 400 ) );
+				}
+				if ( ! isset( $data['timezone'] ) ) {
+					/* translators: %s: Missing parameter name */
+					return new WP_Error( 'missing_required_param', sprintf( __( 'Request parameter is empty: %s.', 'google-site-kit' ), 'timezone' ), array( 'status' => 400 ) );
+				}
+
+				if ( ! $this->authentication->get_oauth_client()->using_proxy() ) {
+					return new WP_Error( 'requires_proxy', __( 'Analytics provisioning requires connecting via the Site Kit proxy.', 'google-site-kit' ), array( 'status' => 400 ) );
```
Comment (felixarntz) [inline includes/Modules/Analytics.php:685]: We don't use the "proxy" term anywhere in user-facing language, let's refer to "Analytics provisioning requires connecting via Site Kit Service."

File: includes/Modules/Analytics.php [hunk 4]
```diff
@@ -1325,29 +1331,10 @@ private function get_analyticsreporting_service() {
...
-						'parameters' => array(),
-					),
-				),
-			)
-		);
 		return array(
 			'analytics'             => new Google_Service_Analytics( $client ),
 			'analyticsreporting'    => new Google_Service_AnalyticsReporting( $client ),
-			'analyticsprovisioning' => $analytics_provisioning_service,
+			'analyticsprovisioning' => new Google_Service_AnalyticsProvisioning( $client, $google_proxy->url() ),
```
Comment (felixarntz) [inline includes/Modules/Analytics.php:1337]: +1 for using this name. Original proposal was `Google_Proxy_Service_AnalyticsProvisioning`, but I think this one works well. Without the proxy, this wouldn't be a separate class anyway and we'd use `Google_Service_Analytics`.

File: includes/Modules/Analytics/proxy-service-analytics-provisioning.php (hunk 1 above)
Comment (adamsilverstein) [reply includes/Modules/Analytics/proxy-service-analytics-provisioning.php]: 👍 Right! Renamed in https://github.com/google/site-kit-wp/pull/1512/commits/25c37c1007a0d9fd912cc8bd6a0088ebdef92c0a.

File: includes/Modules/Analytics/proxy-service-analytics-provisioning.php (hunk 2 above)
Comment (adamsilverstein) [reply includes/Modules/Analytics/proxy-service-analytics-provisioning.php]: Added in https://github.com/google/site-kit-wp/pull/1512/commits/6a9a99a266df117cbc56e26e125b069b0356f6d7.

File: includes/Modules/Analytics.php (hunk 3 above)
Comment (adamsilverstein) [reply includes/Modules/Analytics.php]: Good point! Updated in https://github.com/google/site-kit-wp/pull/1512/commits/721ee981a9a5077e656c047e0923675527a468f2.

File: includes/Modules/Analytics.php (hunk 4 above)
Comment (adamsilverstein) [reply includes/Modules/Analytics.php:1337]: 👍

File: includes/Modules/Analytics/Google_Service_AnalyticsProvisioning.php [hunk 5]
```diff
@@ -24,6 +24,7 @@ class Google_Service_AnalyticsProvisioning extends Google_Service_Analytics {
 	/**
 	 * Constructs the internal representation of the Analytics service.
 	 *
+	 * @since n.e.x.t
```
Comment (felixarntz) [inline includes/Modules/Analytics/Google_Service_AnalyticsProvisioning.php:27]: There should be a blank line above param/return docs. I'll just add that really quick.

Comment (adamsilverstein) [reply includes/Modules/Analytics/Google_Service_AnalyticsProvisioning.php:27]: Thanks :)

**Pull request merged**